src/game_objects/player.cpp

```diff
@@ -1,10 +1,91 @@
 #include "player.h"
 
-// Test code for playing animation for the boy
+#include "asset_loader.h"
+#include "events.h"
+
+Player::Player() : GameObject() {
+    to_moveX = 0;
+    to_moveZ = 0;
+    speed = 2.0f;
+    direction = glm::vec3(0.0f);
+
+    events::RigidBodyData rigid = {
+        glm::vec3(10,0.0f,10), //position
+        1, //mass
+        capsule, //btshape
+        glm::vec3(0,0,0), //inertia
+        this //the gameobject
+    };
+    events::request_rigidbody_event(rigid);
+    
+    // Lock y-axis
+    rigidbody->setLinearFactor(btVector3(1, 0.0f, 1));
+    //Lock angular rotation
+    rigidbody->setAngularFactor(0);
+
+    setup_listeners();
+
+    transform.set_position(10.0f, 0.0f, 10.0f);
+}
+
+void Player::setup_listeners() {
+    events::dungeon::player_move_event.connect([&](events::StickData d) {
+        to_moveX = d.state.first;
+        to_moveZ = d.state.second;
+    });
+
+    events::dungeon::player_interact_event.connect([&]() {
+        interact();
+    });
+
+    events::dungeon::player_raycast_response_event.connect([&](GameObject *bt_hit) {
+        // Second part of the interact function.
+        if (dynamic_cast<Construct*>(bt_hit)) {
```

Comment:
Will this always be a construct?  Also if you're just checking for nullptr, I don't think you need a dynamic cast..?

```cpp
if (bt_hit) {
  dynamic_cast<Construct*>(bt_hit)->interact_trigger();
}
```